pkg/sync/kubernetes/kubernetes_sync.go

```diff
@@ -74,34 +77,130 @@ func (k *Sync) Notify(ctx context.Context, c chan<- sync.INotify) {
 	k.Logger.Infof("Starting kubernetes sync notifier for resource %s", k.ProviderArgs["featureflagconfiguration"])
 	kubeconfig := os.Getenv("KUBECONFIG")
 
-	// Create the client configuration
-	config, err := clientcmd.BuildConfigFromFlags("", kubeconfig)
+	var clusterConfig *rest.Config
+	var err error
+	if kubeconfig != "" {
+		clusterConfig, err = clientcmd.BuildConfigFromFlags("", kubeconfig)
+	} else {
+		clusterConfig, err = rest.InClusterConfig()
+	}
 	if err != nil {
-		k.Logger.Panic(err.Error())
+		k.Logger.Fatalln(err)
 	}
 
-	if k.ProviderArgs["resyncperiod"] != "" {
-		hr, err := time.ParseDuration(k.ProviderArgs["resyncperiod"])
-		if err != nil {
-			k.Logger.Panic(err.Error())
-		}
-		resyncPeriod = hr
+	if err := ffv1alpha1.AddToScheme(scheme.Scheme); err != nil {
+		k.Logger.Panic(err.Error())
 	}
 
-	k.client, err = featureflagconfiguration.NewForConfig(config)
+	k.client, err = client.New(clusterConfig, client.Options{Scheme: scheme.Scheme})
 	if err != nil {
-		k.Logger.Panic(err.Error())
+		k.Logger.Fatalln(err)
 	}
 
-	if err := ffv1alpha1.AddToScheme(scheme.Scheme); err != nil {
-		k.Logger.Panic(err.Error())
+	clusterClient, err := dynamic.NewForConfig(clusterConfig)
+	if err != nil {
+		log.Fatalln(err)
 	}
 
-	go featureflagconfiguration.WatchResources(ctx, *k.Logger.WithFields(log.Fields{
-		"sync":      "kubernetes",
-		"component": "watchresources",
-	}), k.client, resyncPeriod, controllerClient.ObjectKey{
+	resource := ffv1alpha1.GroupVersion.WithResource("featureflagconfigurations")
+	factory := dynamicinformer.NewFilteredDynamicSharedInformerFactory(clusterClient, time.Minute, corev1.NamespaceAll, nil)
+	informer := factory.ForResource(resource).Informer()
+
+	objectKey := client.ObjectKey{
 		Name:      k.ProviderArgs[featureFlagConfigurationName],
 		Namespace: k.ProviderArgs[featureFlagNamespaceName],
-	}, c)
+	}
+	informer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc: func(obj interface{}) {
+			if err := createFuncHandler(obj, objectKey, c); err != nil {
+				k.Logger.Warn(err.Error())
+			}
+		},
+		UpdateFunc: func(oldObj, newObj interface{}) {
+			if err := updateFuncHandler(oldObj, newObj, objectKey, c); err != nil {
+				k.Logger.Warn(err.Error())
+			}
+		},
+		DeleteFunc: func(obj interface{}) {
+			if err := deleteFuncHandler(obj, objectKey, c); err != nil {
+				k.Logger.Warn(err.Error())
+			}
+		},
+	})
+
+	ctx, cancel := signal.NotifyContext(context.Background(), os.Interrupt)
```

Comment:
The top level `context.Context` from `pkg/runtime/start.go` is already listening to `osInterrupt` signals, this context is passed into the start syncer function and down through notify, i believe lines 131 and 132 can be removed.